src/Currencies/AggregateCurrencies.php

```diff
@@ -46,31 +36,24 @@ public function contains(Currency $currency)
         return false;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function subunitFor(Currency $currency)
+    public function subunitFor(Currency $currency): int
     {
         foreach ($this->currencies as $currencies) {
             if ($currencies->contains($currency)) {
                 return $currencies->subunitFor($currency);
             }
         }
 
-        throw new UnknownCurrencyException('Cannot find currency '.$currency->getCode());
+        throw new UnknownCurrencyException('Cannot find currency ' . $currency->getCode());
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getIterator()
+    /** {@inheritDoc} */
+    public function getIterator(): Traversable
     {
-        $iterator = new \AppendIterator();
-
         foreach ($this->currencies as $currencies) {
-            $iterator->append($currencies->getIterator());
+            foreach ($currencies as $currency) {
+                yield $currency;
```

Comment:
Agree here, but isn't this a BC break? From rewindable to non-rewindable iterator?

Reply:
Did not think about this: I can revert the code overall, if preferred? Would need to add some type suppressions, but no biggie.

Reply:
To be honest I am often struggling with this. From a writer and reader perspective generators result in code that is easier to write and to easier to read, but it is not infrequent that I hit a generator that is not rewindable. Let's reverse it here.

Reply:
What do you think? I would prefer a rewindable iterator here.

Reply:
Ah yes, I simply forgot to tackle this comment 👍

Reply:
If you could revert this one, I will merge it in!

Reply:
I did some investigation on this, and it looks kinda broken with `AppendIterator`: https://3v4l.org/sImTa

Is that the expected behavior? :thinking:

Reply:
Nvm, fixed: 8e0630f367f21af20de3221f7068431f05b07c90